pkg/imageverifycache/client.go

```diff
@@ -0,0 +1,128 @@
+package imageverifycache
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/dgraph-io/ristretto"
+	"github.com/go-logr/logr"
+	v1 "github.com/kyverno/kyverno/api/kyverno/v1"
+)
+
+const (
+	defaultTTL     = 1 * time.Hour
+	deafultMaxSize = 1000
+)
+
+type cache struct {
+	logger         logr.Logger
+	isCacheEnabled bool
+	maxSize        int64
+	ttl            time.Duration
+	lock           sync.Mutex
+	Cache          *ristretto.Cache
+}
+
+type Option = func(*cache) error
+
+func New(options ...Option) (Client, error) {
+	cache := &cache{}
+	for _, opt := range options {
+		if err := opt(cache); err != nil {
+			return nil, err
+		}
+	}
+	config := ristretto.Config{
+		MaxCost:     cache.maxSize,
```

Comment:
Isn't Max cost the size in MBs? With max size we are defining the number of entries. 
How do this and Numcounter interact?

Reply:
```
MaxCost is how eviction decisions are made. For example, if MaxCost is 100 and a new item with a cost of 1 increases total cache cost to 101, 1 item will be evicted.

MaxCost can also be used to denote the max size in bytes. For example, if MaxCost is 1,000,000 (1MB) and the cache is full with 1,000 1KB items, a new item (that's accepted) would cause 5 1KB items to be evicted.

MaxCost could be anything as long as it matches how you're using the cost values when calling Set.
```

In our case our each entry will have a fix cost that is 1. So we can keep maxCost same as maxEntries.

And according to the TinyLFU paper the numcounter should be at least 10x of the maxEntries.